File: chatbot/console_chatbot.py
import requests  # Bibliothek für HTTP-Anfragen
import json      # Bibliothek für JSON-Verarbeitung

# Dein persönlicher API-Schlüssel (ersetze "YOUR_API_KEY" durch deinen tatsächlichen Schlüssel)
api_key = "YOUR_API_KEY"

# OpenAI API-Endpunkt für Textgenerierung
endpoint = "https://api.openai.com/v1/chat/completions"

# HTTP-Header mit Content-Type und Autorisierung
headers = {
    "Content-Type": "application/json",
    "Authorization": f"Bearer {api_key}"
}

def get_response(prompt):
    """
    Sendet eine Anfrage an die OpenAI API mit dem übergebenen Prompt und gibt die generierte Antwort zurück.
    """
    # Definiere den Payload mit den Parametern für die Anfrage
    data = {
        "model": "gpt-4o-2024-08-06",  # Hier wird das verwendete Modell angegeben (z.B. text-davinci-003)
        "messages": [{"role": "user", "content": prompt}],  # Der Text, auf den die API reagieren soll
        "max_tokens": 1000,            # Maximale Anzahl an Token in der Antwort
        "temperature": 0.7            # Kreativitätsfaktor der Antwort (0 = deterministisch, 1 = kreativ)
    }
    
    # Sende die POST-Anfrage an den API-Endpunkt
    response = requests.post(endpoint, headers=headers, json=data)
    
    # Überprüfe den Statuscode der Antwort
    if response.status_code == 200:
        result = response.json()
        return result["choices"][0]["message"]["content"].strip()
    else:
        return f"Fehler: {response.status_code} - {response.text}"

def main():
    print("Willkommen zum Chatbot! (Tippe 'exit' zum Beenden)")
    while True:
        user_input = input("Du: ")
        if user_input.lower() == "exit":
            break
        answer = get_response(user_input)
        print("ChatGPT:", answer)

if __name__ == "__main__":
    main()

File: chatbot/typescript/console_chatbot.js
// npm i openai, nicht vergessen!

const OpenAI = require('openai');
const readline = require('readline');

const client = new OpenAI({
    apiKey: "YOUR_API_KEY" // Setze hier deinen OpenAI-API-Schlüssel ein,
});

// Konversationsverlauf initialisieren mit einer Systemnachricht
let conversation = [
    { role: 'system', content: 'Du bist ein hilfreicher Assistent.' }
];

// Erstelle eine Readline-Schnittstelle für die Konsoleneingabe
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    prompt: 'Du: '
});

function startChat() {
    rl.prompt();
    rl.on('line', async (input) => {
        // Falls der Nutzer "exit" eingibt, beenden wir den Chat
        if (input.toLowerCase() === 'exit') {
            console.log('Chatbot wird beendet.');
            rl.close();
            return;
        }
        // Füge die Nutzereingabe zur Konversation hinzu
        conversation.push({ role: 'user', content: input });
        
        try {
            // Anfrage an die API mit dem gesamten Konversationsverlauf
            const chatCompletion = await client.chat.completions.create({
                model: 'gpt-4o', // Verwende hier das gewünschte Modell (z.B. gpt-3.5-turbo oder gpt-4o)
                messages: conversation,
            });
            const answer = chatCompletion.choices[0].message.content;
            console.log('Bot: ' + answer);
            // Füge die Antwort des Assistenten zum Konversationsverlauf hinzu
            conversation.push({ role: 'assistant', content: answer });
        } catch (error) {
            console.error('Fehler:', error.response ? error.response.data : error.message);
        }
        rl.prompt();
    });
}

console.log("Interaktiver Chatbot gestartet. Schreibe deine Nachricht und drücke Enter (Tippe 'exit' zum Beenden).");
startChat();